src/couch_mrview/src/couch_mrview_index.erl

```diff
@@ -226,16 +226,15 @@ verify_index_exists(DbName, Props) ->
         if Type =/= <<"mrview">> -> false; true ->
             DDocId = couch_util:get_value(<<"ddoc_id">>, Props),
             couch_util:with_db(DbName, fun(Db) ->
-                {ok, DesignDocs} = couch_db:get_design_docs(Db),
-                case get_ddoc(DbName, DesignDocs, DDocId) of
-                    #doc{} = DDoc ->
+                case couch_db:get_design_doc(Db, DDocId) of
```

Comment:
This bit here is kind of inconsistent (to be honest it was inconsistent before). We are matching on `not_found`, but clause crashing on `{error, _...`, yet catching and discarding the exception below.

We can either remove this deep `case` all together and always crash on not matching to `{ok, Doc}`, since we are treating errors as `not_found` or go other way around and add `Error -> throw(Error)` clause to make error trace a bit more readable.

Reply:
Hrm. I see your point but I'd think that'd be a second PR if you want to change things. I'm hesitant to change orthogonal logic as part of this work.